### PR TITLE
feat: migrate from sync to async

### PR DESCRIPTION
**Pull request by wswebcreation**
**Contains:**
- replace jasmine for mocha
- change touchPerform to performActions
- support iOS 15
- optimise deep links for iOS simulators

The optimisation for iOS deep links on simulators resulted in a huge improvement

```log
[iPhone 12 iOS 14.5 #0-0] Running: iPhone 12 on iOS 14.5 executing /Users/wimselles/Sauce/Git/appium-boilerplate/apps/iOS-Simulator-NativeDemoApp-0.4.0.app.zip
[iPhone 12 iOS 14.5 #0-0] Session ID: 9dca563b-3124-480f-8779-93c3f1fd4533
[iPhone 12 iOS 14.5 #0-0]
[iPhone 12 iOS 14.5 #0-0] » /tests/specs/app.deep.link.navigation.spec.ts
[iPhone 12 iOS 14.5 #0-0] WebdriverIO and Appium, when navigating by deep link
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the webview
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the login form screen
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the forms screen
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the swipe screen
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the drag and drop screen
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the home screen
[iPhone 12 iOS 14.5 #0-0]
[iPhone 12 iOS 14.5 #0-0] 6 passing (56.6s)
```

Same now with the new way, meaning determine that you are a sim and open it direct from the app with the `browser.url('deeplink://url')` and this is the result

```log
[iPhone 12 iOS 14.5 #0-0] Running: iPhone 12 on iOS 14.5 executing /Users/wimselles/Sauce/Git/appium-boilerplate/apps/iOS-Simulator-NativeDemoApp-0.4.0.app.zip
[iPhone 12 iOS 14.5 #0-0] Session ID: 9dca563b-3124-480f-8779-93c3f1fd4533
[iPhone 12 iOS 14.5 #0-0]
[iPhone 12 iOS 14.5 #0-0] » /tests/specs/app.deep.link.navigation.spec.ts
[iPhone 12 iOS 14.5 #0-0] WebdriverIO and Appium, when navigating by deep link
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the webview
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the login form screen
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the forms screen
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the swipe screen
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the drag and drop screen
[iPhone 12 iOS 14.5 #0-0]    ✓ should be able to open the home screen
[iPhone 12 iOS 14.5 #0-0]
[iPhone 12 iOS 14.5 #0-0] 6 passing (23.9s)
```

Almost half the time!!!!! 🎉

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
 # appium-boilerplate
 
 > **NOTE:**
-> This boilerplate is for Webdriver V7, if you need a boilerplate for:\
-> - V6 please click [here](https://github.com/webdriverio/appium-boilerplate/tree/v6)
-> - V5 please click [here](https://github.com/webdriverio/appium-boilerplate/tree/v5)
-> - V4 please click [here](https://github.com/webdriverio/appium-boilerplate/tree/v4)
+> This boilerplate is for Webdriver V7 where the tests are written in `async` mode and TypeScript.\
```
Comment (christian-bromann) [inline README.md:4]:
```suggestion
> This boilerplate is for Webdriver V7 where the tests are written with `async`/`await` and TypeScript.\
```

File: config/wdio.shared.conf.ts [hunk 2]
```diff
@@ -111,14 +111,14 @@ export const config: WebdriverIO.Config = {
...
-        // Jasmine default timeout
+    // Options to be passed to Mocha.
+    mochaOpts: {
+        ui: 'bdd',
         /**
          * NOTE: This has been increased for more stable Appium Native app
          * tests because they can take a bit longer.
          */
-        defaultTimeoutInterval: 1200000,
+        timeout: 180000,
```
Comment (christian-bromann) [inline config/wdio.shared.conf.ts:121]:
```suggestion
        timeout: 3 * 60 * 1000, // 3min
```

File: tsconfig.json [hunk 3]
```diff
@@ -10,9 +10,11 @@
...
         "types": [
             "node",
-            "webdriverio/sync",
-            "@wdio/jasmine-framework",
-            "expect-webdriverio/jasmine"
+            "webdriverio/async",
+            "@wdio/mocha-framework",
+            "expect-webdriverio",
+            "@wdio/sauce-service",
+            "@wdio/appium-service"
```
Comment (christian-bromann) [inline tsconfig.json:17]: This package doesn't expose types
```suggestion
```

**Review by christian-bromann — APPROVED** [review]
Some minor things otherwise LGTM 👍 

Nice job!

**Comment by wswebcreation** [comment]
Thanks for the review and feedback @christian-bromann

**Pull request merged**